schema/scope_test.go

```diff
@@ -498,3 +498,42 @@ func TestApplyingExternalNamespaceToNonRefTypes(t *testing.T) {
 		})
 	}
 }
+
+func TestMismatchedRoot(t *testing.T) {
```

Comment:
No explicit success case unit test??

Reply:
I don't think the explicit success case is necessary since it is implicitly covered by the rest of the unit tests in `scope_test.go`, but if you want extra credit go for it.

Reply:
Correct. It's called indirectly many times. Many existing tests would break if that wasn't correct.

Reply:
The [comment below](https://github.com/arcalot/arcaflow-plugin-sdk-go/pull/85#pullrequestreview-2009695074) is an example of why we should have (had) an explicit success case.  (That is, there are _four_ ways of doing it wrong and one way of doing it right; it might have been more obvious that we missed one of the wrong ones if we'd had a correct one to compare it to...good catch @mfleader.)

In general, unit tests are intended to test the interface, and they should, to the extent feasible, be complete.  It doesn't matter whether the unit is also used in other units or in functional testing, each unit should also be tested independently.  This allows thing like incorporating mocks into testing with sacrificing coverage.

Reply:
all right I'm convinced

Reply:
Added an explicit success case.